src/NServiceBus.Raw/RawTransportReceiver.cs

```diff
@@ -37,9 +37,9 @@ public void Start()
             isStarted = true;
         }
 
-        public async Task Stop()
+        public async Task Stop(bool force = false)
         {
-            if (!isStarted)
+            if (!isStarted && !force)
```

Comment:
I think we should move this cleanup to a try-catch in the Start method above. So that if IPushMessages.Start above fails, the RawTransportReceiver.Start cleans up before re-throwing the exception. This way the StartableRawEndpoint does not need to care about it.

Reply:
I removed the force flag and added a catch to the Start() call.  It still needs to call Stop() on the Endpoint instance or a connection is leaked.